src/quicklist.c

```diff
@@ -1739,54 +1741,52 @@ int quicklistTest(int argc, char *argv[], int accurate) {
         }
 
         TEST_DESC("add to tail 5x at compress %d", options[_i]) {
-        for (int f = optimize_start; f < 32; f++) {
-                quicklist *ql = quicklistNew(f, options[_i]);
+        for (int f = 0; f < fill_count; f++) {
+                quicklist *ql = quicklistNew(fills[f], options[_i]);
                 for (int i = 0; i < 5; i++)
                     quicklistPushTail(ql, genstr("hello", i), 32);
                 if (ql->count != 5)
                     ERROR;
-                if (f == 32)
+                if (fills[f] == 32)
                     ql_verify(ql, 1, 5, 5, 5);
                 quicklistRelease(ql);
             }
         }
 
         TEST_DESC("add to head 5x at compress %d", options[_i]) {
-            for (int f = optimize_start; f < 32; f++) {
-                quicklist *ql = quicklistNew(f, options[_i]);
+            for (int f = 0; f < fill_count; f++) {
+                quicklist *ql = quicklistNew(fills[f], options[_i]);
                 for (int i = 0; i < 5; i++)
                     quicklistPushHead(ql, genstr("hello", i), 32);
                 if (ql->count != 5)
                     ERROR;
-                if (f == 32)
+                if (fills[f] == 32)
```

Comment:
i guess this was meant to be run in the last iteration (`f==fill_count`), not when the value is 32.
am i wrong?

Reply:
I found out that the original test will never go in, because f will never equal 32.
The original test wanted to verify that the length of quicklist doesn't become 2 when fill>5, and doesn't necessarily need to run to the end, but I think it could be changed to `if (fils[f] >= 5)`